core/src/main/scala/org/broadinstitute/dsde/rawls/snapshot/SnapshotService.scala

```diff
@@ -65,6 +71,19 @@ class SnapshotService(protected val ctx: RawlsRequestContext,
       Future.successful(snapshotRef)
     }
 
+  def validateProtectedStatus(workspaceContext: Workspace, snapshot: NamedDataRepoSnapshot): Unit =
+    // logically it might make more sense to check if the snapshot is protected before the workspace
+    // but that is a more expensive check
+    // check if workspace is protected
+    if (!workspaceContext.bucketName.startsWith("fc-secure")) {
+      // if not, check if snapshot is protected
+      val sources = dataRepoDAO.getSnapshot(snapshot.snapshotId, ctx.userInfo.accessToken).getSource
+      if(sources.exists(_.getDataset.isSecureMonitoringEnabled)) {
+        throw new RawlsException("Unable to add protected snapshot to unprotected workspace.")
+      }
+      }
```

Comment:
maybe a nit: but the close curlies look a bit goofy to me...is this a formatting fix, or an intentional Scala thing?

Reply:
Not a nit!  This actually is the cause of everything breaking all of a sudden!

Reply:
Seconded.